utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -799,12 +799,13 @@ impl<'a, 'b> SyscallsImportsGenerator<'a, 'b> {
 
     /// Reserves enough memory build precise syscall.
     fn reserve_memory(&self) -> i32 {
-        self.memory_size_in_bytes()
-            .saturating_sub(Self::PRECISE_SYSCALL_MEMORY_SIZE) as i32
+        self.memory_size_bytes()
+            .saturating_sub(MemoryLayout::RESERVED_MEMORY_SIZE)
+            .saturating_sub(Self::PRECISE_SYSCALL_RESERVED_MEMORY_SIZE) as i32
```

Comment:
I believe that should be included into `MemoryLayout`.

Reply:
Each precise syscall can use `MemoryLayout` to store some data. But I think in general we just want to allocate memory for the current precise syscall.